src/middlewared/middlewared/plugins/smb_/passdb.py

```diff
@@ -15,229 +20,71 @@ class Config:
-
-        for p in (pdb.stdout.decode()).splitlines():
-            entry = p.split(':')
-            try:
-                pdbentries.append({
-                    'username': entry[0],
-                    'full_name': entry[2],
-                    'uid': entry[1],
-                })
-            except Exception as e:
-                self.logger.debug('Failed to parse passdb entry [%s]: %s', p, e)
-
-        return pdbentries
+    def passdb_list(self, filters, options):
+        """ query existing passdb users """
+        try:
+            return query_passdb_entries(filters or [], options or {})
+        except PassdbMustReinit:
+            self.synchronize_passdb(True).wait_sync(raise_error=True)
+            return query_passdb_entries(filters or [], options or {})
 
     @private
-    async def update_passdb_user(self, user):
-        if user['smbhash'] == user['pdb']:
-            return
-
-        smbpasswd_string = user['smbhash'].split(":")
-        username = user['username']
-        if len(smbpasswd_string) != 7:
-            self.logger.warning("SMB hash for user [%s] is invalid. Authentication for SMB "
-                                "sessions for this user will fail until this is repaired. "
-                                "This may indicate that configuration was restored without a secret "
-                                "seed, and may be repaired by resetting the user password.", username)
-            return
-
-        if user['pdb'] is None:
-            cmd = [SMBCmd.PDBEDIT.value, '-d', '0', '-a', username]
-            rid = db_id_to_rid(IDType.USER, user['id'])
-            cmd.extend(['-U', str(rid)])
-
-            cmd.append('-t')
-
-            pdbcreate = await run(
-                cmd, input=" \n \n".encode(), check=False,
-            )
-            if pdbcreate.returncode != 0:
-                raise CallError(f'{username}: failed to create passdb user: {pdbcreate.stderr.decode()}')
-
-            setntpass = await run(
-                [SMBCmd.PDBEDIT.value, '-d', '0', '--set-nt-hash', smbpasswd_string[3], username], check=False,
-            )
-            if setntpass.returncode != 0:
-                raise CallError(f'Failed to set NT password for {username}: {setntpass.stderr.decode()}')
-
-            if user['locked']:
-                disableacct = await run([SMBCmd.SMBPASSWD.value, '-d', username], check=False)
-                if disableacct.returncode != 0:
-                    raise CallError(f'Failed to disable {username}: {disableacct.stderr.decode()}')
-
-            # Remove any potential negative cache entries
-            try:
-                await self.middleware.call('idmap.gencache.del_idmap_cache_entry', {
-                    'entry_type': 'UID2SID',
-                    'entry': user['uid']
-                })
-            except MatchNotFound:
-                pass
+    def update_passdb_user(self, user: UserEntry):
```

Comment:
Maybe add a comment to clarify usage.

`update_passdb_entry` called below states "Update an existing passdb entry", but this code appears to support an `existing_entry` of None, which suggests adding a **new** entry.  Is this correct?